core/src/executor/delete.rs

```diff
@@ -0,0 +1,73 @@
+use {
+    super::{
+        fetch::{fetch, fetch_columns},
+        ExecuteError, Payload, Referencing,
+    },
+    crate::{
+        ast::{BinaryOperator, Expr, ForeignKey, ReferentialAction},
+        result::{Error, Result},
+        store::{GStore, GStoreMut},
+    },
+    futures::stream::{StreamExt, TryStreamExt},
+    std::rc::Rc,
+};
+
+pub async fn delete<T: GStore + GStoreMut>(
+    storage: &mut T,
+    table_name: &str,
+    selection: &Option<Expr>,
+) -> Result<Payload> {
+    let columns = fetch_columns(storage, table_name).await?.map(Rc::from);
+    let referencings = storage.fetch_referencings(table_name).await?;
+    let keys =
+        fetch(storage, table_name, columns, selection.as_ref())
+            .await?
+            .into_stream()
+            .then(|item| async {
+                let (key, row) = item?;
+
+                for Referencing {
+                    table_name: referencing_table_name,
+                    foreign_key:
+                        ForeignKey {
+                            referencing_column_name,
+                            referenced_column_name,
+                            on_delete,
+                            ..
+                        },
+                } in &referencings
+                {
+                    let value = row.get_value(referenced_column_name).ok_or(
+                        ExecuteError::ValueNotFound(referenced_column_name.to_owned()),
+                    )?;
+
+                    let expr = &Expr::BinaryOp {
+                        left: Box::new(Expr::Identifier(referencing_column_name.clone())),
+                        op: BinaryOperator::Eq,
+                        right: Box::new(value.to_owned().try_into()?),
+                    };
```

Comment:
```rust
let value = row
    .get_value(referenced_column_name)
    .ok_or(ExecuteError::ValueNotFound(
        referenced_column_name.to_owned(),
    ))?
    .clone();

let expr = &Expr::BinaryOp {
    left: Box::new(Expr::Identifier(referencing_column_name.clone())),
    op: BinaryOperator::Eq,
    right: Box::new(Expr::try_from(value)?),
};
```

for value, use of `clone` would be better than `to_owned` because in this case, what we do is converting `&Value` to `Value` which is `&T` to `T`.
we can use to_owned for `&T` to `U` case.